### PR TITLE
fix: preserve all elements when serialized; feat: helper functions for serialization

### LINKED ISSUE #269: feat/write_elements
Hi,

Is there any way to write `List[Element]` data into file and load from it then, in order to avoid to partition data each time?

### LINKED ISSUE #271: Create a data connector for Substack
Create a data connector that:
* fetches one more substack posts
* stores them locally as html files (at least temporarily for processing)
* processes them in a way specfic to substack, i.e. filters out Substack boilerplate elements after calling [partition_html](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/partition/html.py)

See [Adding Data Connectors](https://github.com/Unstructured-IO/unstructured/blob/main/Ingest.md#adding-data-connectors) for details on how to get started. Make sure to include a link to this issue when submitting a PR.

**Definition of Done**

* The [checklist](https://github.com/Unstructured-IO/unstructured/blob/main/Ingest.md#the-checklist) has been completed.
* The connector is able to process a single document or multiple documents (multiple URL's may be passed to `--substack-url` in [main.py](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/ingest/main.py)).
* Note that unlike the [S3IngestDoc](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/ingest/connector/s3_connector.py), the subclass of `BaseIngestDoc` created for this task (probably named `SubstackInstanceDoc`) would need to override the `process_file` method to handle additional cleaning logic. I.e., the 2nd bullet in the [checklist](https://github.com/Unstructured-IO/unstructured/blob/main/Ingest.md#the-checklist).

### PR DESCRIPTION
**Pull request by MthwRobinson**
### Summary

Closes #269 and #270. Fixes `isd_to_elements` so that all element types are preserved when a list of elements is converted to/from a dict. Adds `convert_to_dict`, `dict_to_elements`, and `convert_to_csv` as aliases for functions that use the ISD terminology. Keeps the existing functions for backward compatibility. Add `elements_to_json` and `elements_from_json` for easier serialization and deserialization of `Element` lists.

### Testing

The following workflow should save/load a list of elements, preserving all of them.

```python
    from unstructured.documents.elements import ElementMetadata, Text, Title, FigureCaption
    from unstructured.staging.base import elements_to_json, elements_from_json

    filename = "my-elements.json"
    metadata = ElementMetadata(filename="fake-file.txt")
    elements = [
        FigureCaption(text="caption", metadata=metadata, element_id="1"),
        Title(text="title", metadata=metadata, element_id="2"),
        Text(text="title", metadata=metadata, element_id="3"),

    ]

    elements_to_json(elements, filename=filename)
    new_elements = elements_from_json(filename=filename)
    assert elements == new_elements
```

### CODE REVIEW

File: unstructured/staging/base.py [hunk 1]
```diff
@@ -18,24 +33,67 @@ def convert_to_isd(elements: List[Text]) -> List[Dict[str, str]]:
...
 
-def isd_to_elements(isd: List[Dict[str, str]]) -> List[Text]:
-    """Converts an Initial Structured Data (ISD) dictionary to a list of Text elements."""
-    elements: List[Text] = list()
+def convert_to_dict(elements: List[Element]) -> List[Dict[str, str]]:
+    """Converts a list of elements into a dictionary."""
+    return convert_to_isd(elements)
+
+
+def elements_to_json(elements: List[Element], filename: str):
```
Comment (cragwolfe) [inline unstructured/staging/base.py:41]: please add `indent` a param that gets passed down?

**Comment by cragwolfe** [comment]
Please update https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/ingest/interfaces.py#L94 to use the new method, since that just makes more sense now that we have it, also with `indent=2 😋 .

Would need to push updated json's in https://github.com/Unstructured-IO/unstructured/tree/main/test_unstructured_ingest/expected-structured-output/s3-small-batch/small-pdf-set
after running https://github.com/Unstructured-IO/unstructured/blob/main/test_unstructured_ingest/test-ingest.sh -- it's fine if we have `filename` back in the `metadata` in those fixtures.

File: docs/source/bricks.rst [hunk 2]
```diff
@@ -887,33 +887,33 @@ Staging
...
+  isd = convert_to_dict(elements)
 
 
-``isd_to_elements``
--------------------
+``dict_to_elements``
+---------------------
 
-Converts outputs from initial structured data (ISD) format back to a list of ``Text`` elements.
+Converts a dictionary format back to a list of ``Element`` objects.
```
Comment (qued) [inline docs/source/bricks.rst:910]:
```suggestion
Converts a dictionary of the format produced by ``convert_to_dict`` back to a list of ``Element`` objects.
```

**Review by qued — APPROVED** [review]
1 suggestion, otherwise LGTM pending getting the tests working.

**Review by cragwolfe — APPROVED** [review]

**Pull request merged**